apps/sharedtest/test/unit/l10n/l10n_test.js

```diff
@@ -155,6 +155,23 @@ suite('L10n', function() {
       _translateFragment(elem);
       assert.equal(elem.getAttribute('aria-moz-hint'), 'moz-hint via ARIA');
     });
+
+    test('for a document fragment', function() {
+      var fragment = document.createDocumentFragment();
+      elem.dataset.l10nId = 'textcontent-test';
+      fragment.appendChild(elem);
+      _translateFragment(fragment);
+      assert.equal(elem.textContent, 'this is text content');
```

Comment:
I mean that by design we make all of our DOM API async. The fact that the current master is sync is going away soon: https://github.com/stasm/l20n.js/blob/lightweight/src/bindings/html/dom.js#L36-L40

But since you're right, it will be perma-red, we should probably just fix it when we'll be landing the new API (because it'll have .then which will simplify turning this test async). @stasm, are you ok with that approach?